concurrent/shared/src/main/scala/cats/effect/concurrent/Console.scala

```diff
@@ -0,0 +1,180 @@
+ * for extremely simple console output.
+ *
+ * @example {{{
+ *   import cats.effect.concurrent.Console
+ *   import cats.effect.kernel.Sync
+ *   import cats.syntax.all._
+ *
+ *   def myProgram[F[_]: Sync]: F[Unit] =
+ *     for {
+ *       _ <- Console[F].println("Please enter your name: ")
+ *       n <- Console[F].readLine
+ *       _ <- if (n.nonEmpty) Console[F].println(s"Hello $n!")
+ *            else Console[F].printlnErr("Name is empty!")
+ *     } yield ()
+ * }}}
+ */
+class Console[F[_]] private (val F: Sync[F]) extends AnyVal {
+
+  /**
+   * Reads a line as a string from the standard input by using the platform's default charset, as per
+   * `java.nio.charset.Charset.defaultCharset()`.
+   *
+   * @return an effect that describes reading the user's input from the standard input as a string
+   */
+  def readLine: F[String] =
+    readLineWithCharset(Charset.defaultCharset())
+
+  /**
+   * Reads a line as a string from the standard input by using the provided charset.
+   *
+   * @param charset the `java.nio.charset.Charset` to be used when decoding the input stream
+   * @return an effect that describes reading the user's input from the standard input as a string
+   */
+  def readLineWithCharset(charset: Charset): F[String] =
+    F.interruptible(false) {
+      val in = System.in
+      val decoder = charset
+        .newDecoder()
+        .onMalformedInput(CodingErrorAction.REPORT)
+        .onUnmappableCharacter(CodingErrorAction.REPLACE)
+      val bytes = ByteBuffer.allocate(16)
+      val builder = new JStringBuilder()
+
+      def decodeNext(): CharBuffer = {
+        bytes.clear()
+        decodeNextLoop()
+      }
+
+      @tailrec
+      def decodeNextLoop(): CharBuffer = {
+        val b = in.read()
+        if (b == -1) {
+          return null
+        }
+        bytes.put(b.toByte)
+        val limit = bytes.limit()
+        val position = bytes.position()
+        try {
+          bytes.flip()
+          decoder.decode(bytes)
+        } catch {
+          case _: MalformedInputException =>
+            bytes.limit(limit)
+            bytes.position(position)
+            decodeNextLoop()
+        }
+      }
+
+      @tailrec
+      def loop(): String = {
+        val decoded = decodeNext().toString()
+        builder.append(decoded)
+        println(decoded)
+        if (decoded == "\n") {
+          val len = builder.length()
+          if (len > 0) {
+            if (builder.charAt(len - 1) == '\r') {
+              builder.deleteCharAt(len - 1)
+            }
+          }
+          builder.toString()
+        } else loop()
+      }
+
+      loop()
+    }
+
+  /**
+   * Prints a value to the standard output using the implicit `cats.Show` instance.
+   *
+   * @param a value to be printed to the standard output
+   */
+  def print[A: Show](a: A): F[Unit] =
+    F.blocking(System.out.print(a.show))
```

Comment:
Relatively minor niggle, but `a.show` should be run *outside* of the `blocking` pool.

```suggestion
  def print[A: Show](a: A): F[Unit] = {
    val text = a.show
    F.blocking(System.out.print(text))
  }
```

Same comment for the other `print`s.

Reply:
Nice. Done.